WcaOnRails/app/models/delegate_report.rb

```diff
@@ -23,9 +23,15 @@ def equipment_default
 Gen 3 Display: 0"
   end
 
+  validates :schedule_url, presence: true, if: :schedule_and_disussion_urls_required?
   validates :schedule_url, url: true
+  validates :discussion_url, presence: true, if: :schedule_and_disussion_urls_required?
   validates :discussion_url, url: true
 
+  def schedule_and_disussion_urls_required?
+    posted? && created_at.present? && created_at > Date.new(2019, 7, 21)
```

Comment:
Nit: we don't need the `created_at.present?` check as the `delegate_reports` table has `NOT NULL` on `created_at` (it was necessary for comps, as we started to keep track of creation when Rails kicked in).

Reply:
Updated! I'll merge this one as soon as tests pass.